include/ardpolyfills/WString.h

```diff
@@ -83,52 +83,126 @@ class String {
         return true;
     }
 
+    /**
+    * Compares two Strings, testing whether one comes before or after the other, or whether they�re equal
+    **/
     [[nodiscard]] int compareTo(const String& s) const noexcept;
 
+    /**
+    * Tests whether or not a String starts with the characters of another String
+    **/
     [[nodiscard]] bool startsWith(const String& s) const noexcept;
 
+    /**
+    * Tests whether or not a String ends with the characters of another String
+    **/
     [[nodiscard]] bool endsWith(const String& s) const noexcept;
 
+    /**
+    * Copies the String�s characters to the supplied buffer
+    **/
     void getBytes(byte* buffer, unsigned length) const noexcept;
 
+    /**
+    * Locates a character within a string
+    **/
     [[nodiscard]] int indexOf(const char* c) const noexcept;
 
+    /**
+    * Locates a character within a string but begins the search from given index
+    **/
     [[nodiscard]] int indexOf(const char* c, unsigned index) const noexcept;
 
+    /**
+    * Locates String within another String
+    **/
     [[nodiscard]] int indexOf(const String& str) const noexcept;
 
+    /**
+    * Locates String within another String but begins search from a given index
+    **/
     [[nodiscard]] int indexOf(const String& str, unsigned index) const noexcept;
 
+    /**
+    * Removes a char from from a string at given index
+    **/
     void remove(unsigned idx);
 
+    /**
+    * Removes a char from a string at given index to index plus count
+    **/
     void remove(unsigned idx, unsigned count);
 
+    /**
+    * Replace substrings of a String with a different substring
+    **/
     void replace(const String& substring1, const String& substring2);
 
+    /**
+    * allocate a buffer in memory for manipulating Strings with given size
+    **/
     void reserve(unsigned size);
 
+    /**
+    * Sets a character of the String at given index
+    **/
     void setCharAt(unsigned index, char c);
 
+    /**
+    * Get a substring of a String from given index
+    **/
     [[nodiscard]] String substring(unsigned from) const;
 
+    /**
+    * Get a substring of a String from given index to givn index
+    **/
     [[nodiscard]] String substring(unsigned from, unsigned to) const;
 
+    /**
+    * Copies the String�s characters to the supplied buffer
+    **/
     void toCharArray(char* buffer, unsigned length) noexcept;
 
+    /**
+    * Converts a valid String to an integer 
+    **/
     [[nodiscard]] long toInt() const noexcept;
 
+    /**
+    * Converts a valid String to a double
+    **/
     [[nodiscard]] double toDouble() const noexcept;
 
+    /**
+    * Converts a valid String to a float
+    **/
     [[nodiscard]] float toFloat() const noexcept;
 
+    /**
+    * Get a lower-case version of a String
+    **/
     void toLowerCase() noexcept;
 
+    /**
+    * Get a upper-case version of a String
```

Comment:
Method doesn't return anything, it transforms the string on which this was called to uppercase